mlflow/entities/base_tag.py

```diff
@@ -0,0 +1,39 @@
+import logging
+import os
+
+from mlflow.entities._mlflow_object import _MLflowObject
+from mlflow.utils.validation_common import MAX_TAG_VAL_LENGTH
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseTag(_MLflowObject):
+    """Base Tag object."""
+
+    def __init__(self, key: str, value: str) -> None:
+        self._key = key
+        auto_truncate = os.environ.get("MLFLOW_AUTO_TRUNCATE_LONG_TAG_VALUE", False)
+        if auto_truncate and len(value) > MAX_TAG_VAL_LENGTH:
+            logger.warning("Long Tag value for key %s is auto truncated.", key)
+            self._value = value[: int(auto_truncate)]
```

Comment:
if `auto_truncate` is False, and `len(value) > MAX_TAG_VAL_LENGTH`, we should raise error right ?

Reply:
if auto_truncate is False we do nothing here.
No error must be raised.
Main idea - error will be raised on validation as it works now.